src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/DocumentService.kt

```diff
@@ -167,6 +167,6 @@ class DocumentService(
     documentEntity.deletionDate = LocalDateTime.now()
     documentsRepository.save(documentEntity)
 
-    return documentsRepository.findFirstByNomsIdAndCategory(nomsId, category, true)
+    return documentsRepository.findByPrisonerIdAndId(prisoner.id!!.toLong(), documentEntity.id!!.toLong())
```

Comment:
we could just return the the result from documentsRepository.save, and avoid extra call to the db. Also I'd argue that a delete call doesn't really need a response body, if you'd rather just remove this

Reply:
done!